File: SameDayDelivery-Admin/model/notification.model.js
const mongoose = require('mongoose');

const notificationModel = new mongoose.Schema({
    notification_id:{
        type: mongoose.Schema.Types.ObjectId,
        required: true,
        default:new mongoose.Types.ObjectId,
    },
    order_id: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "Order",
        required: true,
    },
    notification_type: {
        type: String,
        required: true,
        enum:["order","message","feedback"],
        default:"order",
    },
    notification_title: {
        type: String,
        required: true,
    },
    notification_description: {
        type: String,
        required: true,
    },
    notification_date: {
        type: Date,
        required: true,
        default: Date.now,
    },
});

const Notification = mongoose.model("Notification", notificationModel);
module.exports = Notification;

File: same_day_delivery_client/lib/model/notification.model.dart
class NotificationModel {
  String notificationId;
  String orderId;
  String notificationType;
  String notificationTitle;
  String notificationDescription;
  DateTime notificationDate;

  NotificationModel({
    required this.notificationId,
    required this.orderId,
    required this.notificationType,
    required this.notificationTitle,
    required this.notificationDescription,
    required this.notificationDate,
  });

  factory NotificationModel.fromJson(Map<String, dynamic> json) {
    return NotificationModel(
      notificationId: json['notificationId'],
      orderId: json['orderId'],
      notificationType: json['notificationType'],
      notificationTitle: json['notificationTitle'],
      notificationDescription: json['notificationDescription'],
      notificationDate: DateTime.parse(json['notificationDate']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'notificationId': notificationId,
      'orderId': orderId,
      'notificationType': notificationType,
      'notificationTitle': notificationTitle,
      'notificationDescription': notificationDescription,
      'notificationDate': notificationDate.toIso8601String(),
    };
  }
}
